Formula/cmake.rb

```diff
@@ -29,7 +29,7 @@ class Cmake < Formula
   uses_from_macos "ncurses"
 
   on_linux do
-    depends_on "openssl@1.1"
+    depends_on "openssl@3"
   end
 
   # The completions were removed because of problems with system bash
```

Comment:
Unrelated to this PR, just noticed this comment. I previously moved completions but may want to revert that.

Personally, I haven't seen original issue when using system `bash` 3.2.57 with `bash-completion`. I noticed that running:
```sh
source "$HOMEBREW_PREFIX/etc/profile.d/bash_completion.sh"
```
automatically enables:
```console
❯ shopt extglob
extglob        	on
```

The errors in original issue https://github.com/Homebrew/legacy-homebrew/issues/39083 would happen if `extglob` is not enabled, which could happen if user doesn't use `bash-completion` and directly sources completions.

Since we do document that approach (Ref: https://docs.brew.sh/Shell-Completion#configuring-completions-in-bash), it may be better to revert change to install completions into `bash-completion@2` directory instead.